### PR TITLE
Added oblique resolutions to caminfo and the advanced track tool of qview and qnet. Fixes #4100. 

### LINKED ISSUE #4100: Issues updating isis4 (isis4.0.1 to isis4.3.0)
Here is what I get when I fire up my conda:

**4.0.1 # Public version number
2020-03-04 # Release date
stable # Release stage (alpha, beta, stable)**

Next I try to update isis
**> conda update isis**

I get back:
**Collecting package metadata (repodata.json): done
Solving environment: done**

**All requested packages already installed.**

**4.0.1 # Public version number
2020-03-04 # Release date
stable # Release stage (alpha, beta, stable)**

I did some basic debugging:
**>conda config --show channels**
**channels:**
  **- usgs-astrogeology**
  **- conda-forge**
  **- defaults**

Looks OK so I tried to point straight to isis4.3.0
**>conda install isis4.3.0
Collecting package metadata (repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.**

**PackagesNotFoundError: The following packages are not available from current channels:**

  **- isis4.3.0**

**Current channels:**

  - https://conda.anaconda.org/usgs-astrogeology/linux-64
  - https://conda.anaconda.org/usgs-astrogeology/noarch
  - https://conda.anaconda.org/conda-forge/linux-64
  - https://conda.anaconda.org/conda-forge/noarch
  - https://repo.anaconda.com/pkgs/main/linux-64
  - https://repo.anaconda.com/pkgs/main/noarch
  - https://repo.anaconda.com/pkgs/r/linux-64
  - https://repo.anaconda.com/pkgs/r/noarch

**To search for alternate channels that may provide the conda package you're
looking for, navigate to**

    **https://anaconda.org**

**and use the search bar at the top of the page.**

I also tried
**>conda install -c usgs-astrogeology isis=4.3.0**

with the same results.

I made sure my conda is the latest version.
I'm out of ideas...  have any for me?

If I do a fresh isis4 install, I wanted to confirm that you are still on python 3.6?

Thanks!
Laz

### PR DESCRIPTION
**Pull request by kaitlyndlee**
The advanced track tool used to use an enumeration to get the index where each element should be added to to the track tool. When adding the oblique resolutions, I accidentally did not add the resolutions in the same order as they were in the enumeration. To fix this, in the header file, I created a QList that stores the elements we want to add. Then, in the cpp file, I added a loop that adds each item from the QList to the left-hand side of the track tool. I then implemented a method call getIndex() that calculates where each element should be added to the track tool. This way, the index does not depend on the enumeration.

### CODE REVIEW

File: isis/src/base/apps/caminfo/caminfo.xml [hunk 1]
```diff
@@ -20,191 +20,189 @@
...
-    Bands       = 1
-  End_Object
+<pre>
+  Object = Caminfo
+    Object = Parameters
+      Program     = caminfo
+      IsisVersion = "3.5.2.0 beta | 2017-11-04"
+      RunDate     = 2018-03-30T20:41:47
+      IsisId      = MeSSEnGeR/MDIS-WAC/1/0131773041:923000
+      From        = EW0131773041G.cal.cub
```
Comment (celbal) [inline isis/src/base/apps/caminfo/caminfo.xml:45]: Why did you choose to change the example?

File: isis/src/base/apps/caminfo/caminfo.xml [hunk 2]
```diff
@@ -362,6 +360,11 @@ End
     <change name="Janet Richie" date="2013-02-25">
       Reviewed documentation. References #1452.
     </change>
+    <change name="Kaitlyn Lee" date="2018-03-06">
+      Added oblique sample, line, detector, and pixel resolutions to the
+      geometry PVL group. Updated example output in application description,
+      so that the addded oblique elements are included.Fixes #4100.
```
Comment (celbal) [inline isis/src/base/apps/caminfo/caminfo.xml]: Need a space between include.Fixes

File: isis/src/qisis/objs/AdvancedTrackTool/AdvancedTrackTool.cpp [hunk 3]
```diff
@@ -61,58 +63,40 @@ namespace Isis {
...
-    p_tableWin->addToTable(false, "Serial Number", "Serial Number");
-    p_tableWin->addToTable(false, "Track Mosaic Index", "Track Mosaic Index");
-    p_tableWin->addToTable(false, "Track Mosaic FileName", "Track Mosaic FileName");
-    p_tableWin->addToTable(false, "Track Mosaic Serial Number", "Track Mosaic Serial Number");
-    p_tableWin->addToTable(false, "Notes", "Notes");
-    //This variable will keep track of how many times
+    // Adds each item of checkBoxItems to the table.
+    // If a tool tip is specified, we cannot skip parameters, so -1 and
+    // Qt::Horizontal are specified.
+    QList< QList<QString> >::iterator iter;
```
Comment (celbal) [inline isis/src/qisis/objs/AdvancedTrackTool/AdvancedTrackTool.cpp:69]: Nice refactor

File: isis/src/qisis/objs/AdvancedTrackTool/AdvancedTrackTool.cpp [hunk 4]
```diff
@@ -220,27 +204,55 @@ namespace Isis {
...
     }
   }
 
+    /**
+     * This method finds the index of the header in checkBoxItems by looping
+     * through checkBoxItems, grabbing the header from each QList, and parsing
+     * the header at ":" to account for check boxes turning on multiple columns
+     *
+     * @param keyword Header to be found
+     * @return int
```
Comment (celbal) [inline isis/src/qisis/objs/AdvancedTrackTool/AdvancedTrackTool.cpp]: Describe what the int is

File: isis/src/qisis/objs/AdvancedTrackTool/AdvancedTrackTool.h [hunk 5]
```diff
@@ -82,6 +85,12 @@ namespace Isis {
    *                          before it attempts to record a point so that a table is created
    *                          to record the point into so that the first recorded point is drawn.
    *                          Fixes #5143.
+   *  @history 2018-03-07 Kaitlyn Lee - Added columns for oblique pixel, sample, line,
+`   *                         and detector resolutions. Added checkBoxItems and loop to add
```
Comment (celbal) [inline isis/src/qisis/objs/AdvancedTrackTool/AdvancedTrackTool.h]: * out of line

**Comment by celbal** [comment]
Did you run the tests that Tammy had commented on the ticket?

**Comment by celbal** [comment]
Make sure Tyler tests this as well.

File: isis/src/base/apps/caminfo/caminfo.xml (hunk 1 above)
Comment (twilson271828) [reply isis/src/base/apps/caminfo/caminfo.xml:45]: I told her to do that.  She was having difficulty finding the original cube that was used in the example, so I pointed her to another Messenger cube.

**Comment by kaitlyndlee** [comment]
The tests that Tammy had made previously were testing ticket #476 that Tyler was working on. #4100 is adding what Tyler did for #476 to caminfo and qview/qnet.

**Comment by twilson271828** [comment]
The tests Tammy refers to were the additions I made to mosrange/campt/phocube/camstats previous to this ticket when I fixed m00476.

**Comment by celbal** [comment]
So you propagated code from one app to other apps?

**Comment by kaitlyndlee** [comment]
If I understand what you mean correctly, yes. For ticket #4100, I am adding the oblique resolutions Tylor calculated with ticket #476. The tests Tammy is referring to are the additions Tylor made to mosrange/campt/phocube/camstats when he worked on ticket #476.

**Comment by celbal** [comment]
so it means that the resolutions that were added to these apps should be tested in the corresponding app tests. Or else you aren't going to get 100% code coverage.

**Comment by kaitlyndlee** [comment]
The changes made to caminfo will just add four entries to the Geometry label of the PVL files. Because the tests already add the Geometry label to the output PVL files and test this feature, all that is needed is to checkin the new test data. For qview and qnet, the tests are still passing.

**Review by celbal — APPROVED** [review]

**Pull request merged**